### PR TITLE
New postgresql cheatsheet

### PR DESCRIPTION
**Pull request by gautamkrishnar**
New postgresql cheatsheet

idea: https://duck.co/ideas/idea/6188/postgresql-cheat-sheet

**Requires suggestions for more content....**

------
ia: https://duck.co/ia/view/postgresql_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+  "id": "postgresql_cheat_sheet",
+  "name": "Postgre SQL",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/postgresql.json]: @gautamkrishnar lets remove the space. Its "PostgreSQL"

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  "description": "PostgreSQL, often simply Postgres, is an object-relational database management system",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/postgresql.json]: @gautamkrishnar I  think some thing like 
"PostgreSQL is an open source object-relational database management system" is better

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      },
+      {
+        "key": "CREATE DATABASE mypgdatabase OWNER mypguser;",
+        "val": "Create a database"
+      }
+    ],
+    "Postgresql service": [
+      {
+        "key": "//etc//init.d//postgresql reload",
+        "val": "reload postgresql"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/postgresql.json]: @gautamkrishnar  capitalise r

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  "metadata": {
+    "sourceName": "Debian wiki",
+    "sourceUrl": "https://wiki.debian.org/PostgretionSql"
+  },
+  "section_order": [
+    "Installation",
+    "User access",
+    "New User and database",
+    "Postgresql service",
+    "Listing existing database clusters"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/postgresql.json]: @gautamkrishnar  I think "List" is better than "Listing"

Comment (jophab) [inline share/goodie/cheat_sheets/json/postgresql.json]: @gautamkrishnar also section names would be better if
"User Access"
"New User and Database"
"Postgresql Service"
"Listing existing Database Clusters"

are used.

**Comment by jophab** [comment]
@gautamkrishnar This LGTM :+1:

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 5]
```diff
@@ -0,0 +1,69 @@
+{
+  "id": "postgresql_cheat_sheet",
+  "name": "PostgreSQL",
+  "description": "PostgreSQL is an object-relational database management system",
+  "template_type": "terminal",
+  "metadata": {
+    "sourceName": "Debian wiki",
+    "sourceUrl": "https://wiki.debian.org/PostgretionSql"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/postgresql.json]: This url doesn't exist :)

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  },
+  "section_order": [
+    "Installation",
+    "User Access",
+    "New User and Database",
+    "Postgresql Service",
+    "List Existing Database Clusters"
+  ],
+  "sections": {
+    "Installation": [
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/postgresql.json]: This installation section only applies to Debian based systems

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    ],
+    "User Access": [
+      {
+        "key": "psql",
+        "val": "Connect to the database"
+      },
+      {
+        "key": "psql -d mypgdatabase -U mypguser",
+        "val": "Connect as user mypguser to mypgdatabase"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/postgresql.json]: Can we change this to: `Connect to localhost as user ` or include the `-h` for host?

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        "val": "Create a user"
+      },
+      {
+        "key": "CREATE DATABASE mypgdatabase OWNER mypguser;",
+        "val": "Create a database"
+      }
+    ],
+    "Postgresql Service": [
+      {
+        "key": "//etc//init.d//postgresql reload",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/postgresql.json]: With upstart (Ubuntu) and systemd (Debian 8+) the preferred method is `sudo service postgresql reload`

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      {
+        "key": "CREATE USER mypguser WITH PASSWORD 'mypguserpass';",
+        "val": "Create a user"
+      },
+      {
+        "key": "CREATE DATABASE mypgdatabase OWNER mypguser;",
+        "val": "Create a database"
+      }
+    ],
+    "Postgresql Service": [
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/postgresql.json]: Maybe `PostgreSQL Service Management` ?

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+      {
+        "key": "//etc//init.d//postgresql reload",
+        "val": "Reload postgresql"
+      }
+    ],
+    "Listing Existing Database Clusters": [
+      {
+        "key": "pg_lsclusters",
+        "val": "Check installed clusters and obtain some basic informations"
+      }
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/postgresql.json]: Can we add more PG specific things here? `pg_ctrlcluster` `pg_dropcluster` `pg_createcluster` etc..

File: share/goodie/cheat_sheets/json/postgresql.json [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  "template_type": "terminal",
+  "metadata": {
+    "sourceName": "Debian wiki",
+    "sourceUrl": "https://wiki.debian.org/PostgretionSql"
+  },
+  "section_order": [
+    "Installation",
+    "User Access",
+    "New User and Database",
+    "Postgresql Service",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/postgresql.json]: Also can this be stylised into `PostgreSQL` ?

Comment (jophab) [inline share/goodie/cheat_sheets/json/postgresql.json]: @mintsoft @gautamkrishnar :+1: 
Yup that will be better .

**Comment by moollaza** [comment]
I've updated the ID and added some example queries. Screenshots should work now once it's deployed again.

> Requires suggestions for more content....

@gautamkrishnar we can always start small and let the community add to this once it goes live :+1: 

I think this is merge-able

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=postgresql cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/postgresql_cheat_sheet_mobile.png)

**Comment by gautamkrishnar** [comment]
@moollaza thanks :dancer:

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Pull request merged**